src/camera.c

```diff
@@ -1675,26 +1678,22 @@ void update_camera_for_object(Camera *camera)
 }
 #endif
 
-// regalloc
-#ifndef NON_MATCHING
-#pragma GLOBAL_ASM("asm/nonmatchings/camera/transform_point_by_object_matrix.s")
-#else
-void transform_point_by_object_matrix(Vec3f *v, Vec3f *ov, s8 matrixIdx)
-{
-    if (matrixIdx < 0)
-    {
-        ov->x = v->x;
-        ov->y = v->y;
-        ov->z = v->z;
-    }
-    else
-    {
-        vec3_transform(&gObjectMatrices[matrixIdx], 
-            v->x, v->y, v->z, 
-            &ov->x, &ov->y, &ov->z);
-    }
+void transform_point_by_object_matrix(Vec3f* v, Vec3f* ov, s8 matrixIdx) {
+    u32 shouldReturnOriginal = matrixIdx < 0; 
+    
+    if (matrixIdx >= 30) {
```

Comment:
Interesting that this matches, I found different ways to match camera stuff here https://github.com/zestydevy/dinosaur-planet/pull/112/commits/c0bfd62b569d007c39433440cdbdbf603c754d7d